### PR TITLE
support rlhf

### PR DESCRIPTION
**Pull request by Jian1273**
1. Add rm trainer
2. Add rm data processing method
3. Add some configuration parameters to support rm

### CODE REVIEW

**Comment by csunny** [comment]
CI pipeline failed.  Use the following command to fix this.
```
pip install black
black .
```

**Comment by wangzaistone** [comment]
@Jian1273  checks not passed

**Comment by qidanrui** [comment]
@wangzaistone Except the black check, did the submitted code passes through the test? If yes, I think we can first merge it and run black by ourselves later on.

**Comment by csunny** [comment]
> @wangzaistone Except the black check, did the submitted code passes through the test? If yes, I think we can first merge it and run black by ourselves later on.

Yes, But first do CR and tests.  @wangzaistone + @junewgl

**Comment by junewgl** [comment]
OK，I'm going to test the code @csunny

**Comment by junewgl** [comment]
> OK，I'm going to test the code @csunny

After my testing, the code can run successfully, but it needs to add a data set: [`oaast_rm_zh.json`](https://github.com/hiyouga/LLaMA-Factory/blob/main/data/oaast_rm_zh.json)
image

After running, the weights will be saved in the output directory, as shown below:
image

cc @csunny @wangzaistone @qidanrui 


Thanks for your contribution @Jian1273  and I have some `suggestions`：
1.You can consider adding an effect `evaluation code` for the RM model, such as [Microsoft's RLHF code](https://github.com/microsoft/DeepSpeedExamples/tree/master/applications/DeepSpeed-Chat)
2.You can take the time to submit the `RL` training related code to verify the overall RLHF effect.
3.After you submit all the RLHF code, we can test it on the `text2sql` task together.

**Comment by qidanrui** [comment]
I've rerun the CI and the code passes all checks. @junewgl @Jian1273 @csunny @wangzaistone

File: dbgpt_hub/data_process/data_utils.py [hunk 1]
```diff
@@ -690,9 +775,17 @@ def print_supervised_dataset_example(example):
...
-    dataset = dataset.filter(lambda example: example["prompt"] and example["response"])
-    preprocess_function = preprocess_supervised_dataset
-    print_function = print_supervised_dataset_example
+    if stage == "pt":
+        pass
+    elif stage == "sft" and not training_args.predict_with_generate:
+        preprocess_function = preprocess_supervised_dataset
+        print_function = print_supervised_dataset_example
+    elif stage == "rm":
+        print(111111111111111111)
```
Comment (oushu1zhangxiangxuan1) [inline dbgpt_hub/data_process/data_utils.py:784]: debug code remains

**Review by wangzaistone — APPROVED** [review]

File: dbgpt_hub/data_process/data_utils.py (hunk 1 above)
Comment (wangzaistone) [reply dbgpt_hub/data_process/data_utils.py:784]: @junewgl  what 's the detail about the bug  and your command ,env  ?  I have passed to

**Pull request merged**